### PR TITLE
Add Lookup API integration for user details

### PR DESCRIPTION
**Pull request by matiasilva**
* Adds a util for fetching data from Lookup based on a CRSid
* Implements this in the soc registration form

Touches on #2 but to fully implement this, we need to add better methods for querying a list. We can choose to use the full SDK provided by UIS.

### CODE REVIEW

File: lightbluetent/utils.py [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
+import requests
+
 
 def gen_unique_string():
     return str(uuid.uuid4()).replace("-", "")
 
 
+def fetch_lookup_data(crsid):
+    url = f"https://www.lookup.cam.ac.uk/api/v1/person/crsid/{crsid}"
+    res = requests.get(url, params={'fetch', "email,departingEmail"}, timeout=(0.5, 10))
```
Comment (CHTJonas) [inline lightbluetent/utils.py:11]:
```suggestion
    res = requests.get(url, params={"fetch": "email,departingEmail", "format": "json"}, timeout=(0.5, 10))
```

File: lightbluetent/templates/home/register.html [hunk 2]
```diff
@@ -3,9 +3,9 @@
 {% block body %}
   <form method="post">
     <label for="name">Your name</label>
-    <input name=name id="name" value="Default name from lookup" required><br />
+    <input name=name id="name" value="{% if user_data %} user_data.visibleName {% endif %}" required><br />
```
Comment (CHTJonas) [inline lightbluetent/templates/home/register.html:6]:
```suggestion
    <input name=name id="name" value="{% if user_data %} user_data["visibleName"] {% endif %}" required><br />
```

File: lightbluetent/templates/home/register.html [hunk 3]
```diff
@@ -3,9 +3,9 @@
...
     <label for="email_address">Email address</label>
-    <input name=email_address id="email_address" value="{{ crsid }}@cam.ac.uk" required><br />
+    <input name=email_address id="email_address" value="{% if user_data %} user_data.attributes[0]["value"] {% endif %}" required><br />
```
Comment (CHTJonas) [inline lightbluetent/templates/home/register.html:8]:
```suggestion
    <input name=email_address id="email_address" value="{% if user_data %} user_data["attributes"][0]["value"] {% endif %}" required><br />
```

**Review by CHTJonas — CHANGES_REQUESTED** [review]
Nice work! Just a few little suggestions left inline - could also use a rebase on top of the most recent `master`.

File: lightbluetent/utils.py (hunk 1 above)
Comment (matiasilva) [reply lightbluetent/utils.py]: Fixed!

File: lightbluetent/templates/home/register.html (hunk 2 above)
Comment (matiasilva) [reply lightbluetent/templates/home/register.html]: Darn JavaScript habits..

**Review by doismellburning — APPROVED** [review]
Looks good - as you say, I think it would benefit from SDK integration with (presumably) better error handling etc. but looks like it does the job nicely

**Comment by matiasilva** [comment]
Closed in favor of #87.

**Pull request closed without merging**